api/service/src/main/java/com/coreeng/supportbot/teams/PlatformTeamsService.java

```diff
@@ -62,19 +82,85 @@ void init() {
                     ));
                 users.add(user);
             }
+            fetchedGroupUsers.put(groupRef, users);
+        }
 
-            groupIdToUsers.put(t.groupRef(), users);
-            for (PlatformUser user : users) {
-                team.users().add(user);
-                user.teams().add(team);
+        groupIdToUsers.putAll(fetchedGroupUsers);
+        for (PlatformTeam team : teamByName.values()) {
+            for (String groupRef : team.groupRefs()) {
+                List<PlatformUser> users = groupIdToUsers.getOrDefault(groupRef, List.of());
+                team.users().addAll(users);
+                for (PlatformUser user : users) {
```

Comment:
Will this make loading the UI slow at all as we're iterating over 3 loops here? Is there a way we can avoid this?

Not an issue but as the teams grow, we'll start to see performance issues